### PR TITLE
  Add state to disable or enable submit form in DataAdapterForm

### PR DESCRIPTION
**Pull request by ousmaneo**
These changes add a new `isFormExternalyValidated`  state to  `DataAdapterForm` to allow children component to control (disable or enable) the form submit button.

/jenkins-pr-deps Graylog2/graylog-plugin-cloud#617

## Motivation and Context

For the CSV File upload plugin, this allows the submit button to be disabled when the file is being uploaded.
## Types of changes

- [] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by edmundoa — CHANGES_REQUESTED** [review]
The change looks fine to me but I think the naming could be improved. I think the prop `isFormExternalyValidated` and even more `setIsFormExternalyValid` are a bit confusing and do not represent what we want to accomplish with the change, which is letting a children component disable form submission. How about renaming props to `isFormDisabled` and `setIsFormDisabled` respectively?

**Comment by ousmaneo** [comment]
Thank you for reviewing @edmundoa. I have updated the naming.

File: graylog2-web-interface/src/components/lookup-tables/DataAdapterForm.jsx [hunk 1]
```diff
@@ -357,7 +363,7 @@ class DataAdapterForm extends React.Component {
               <fieldset>
                 <Row>
                   <Col mdOffset={3} md={9}>
-                    <Button type="submit" bsStyle="success">{create ? 'Create Adapter'
+                    <Button type="submit" bsStyle="success" disabled={!isFormDisabled}>{create ? 'Create Adapter'
```
Comment (Gaya) [inline graylog2-web-interface/src/components/lookup-tables/DataAdapterForm.jsx:366]: I like the `isFormDisabled` name better as @edmundoa suggested, but right here it's telling the opposite is true. If the form is not not disable, it's disabled. I think the default and this check need to be flipped.

Other than that: looks good to me!

Comment (ousmaneo) [reply graylog2-web-interface/src/components/lookup-tables/DataAdapterForm.jsx:366]: Thank you ! fixed it.

**Review by Gaya — APPROVED** [review]
LGTM!

File: graylog2-web-interface/src/components/lookup-tables/DataAdapterForm.jsx [hunk 2]
```diff
@@ -128,6 +129,10 @@ class DataAdapterForm extends React.Component {
     }
   };
 
+  _setIsFormDisabled = (formValidationState) => {
```
Comment (edmundoa) [inline graylog2-web-interface/src/components/lookup-tables/DataAdapterForm.jsx:132]: Small nitpick: we still use `formValidationState` as argument here.

**Review by edmundoa — APPROVED** [review]
Looks good to me. I added a small nitpick as comment, just in case you want to take care of it. Otherwise, feel free to merge 👍

**Pull request merged**